### ISSUE TITLE
Compile-time c.eval causes java.lang.SecurityException: Prohibited package name: java.util

### ISSUE THREAD

**Issue by olafurpg** [scala-native/scala-native#901]
While trying to port fastparse to native, I hit on a compiler crash during a macro expansion that uses `c.eval`

```scala
[error]     exception during macro expansion:
[error]     scala.tools.reflect.ToolBoxError: reflective compilation has failed: cannot initialize the compiler due to java.lang.SecurityException: Prohibited package name: java.util
[error]     	at scala.tools.reflect.ToolBoxFactory$ToolBoxImpl$withCompilerApi$api$.liftedTree1$1(ToolBoxFactory.scala:345)
[error]     	at scala.tools.reflect.ToolBoxFactory$ToolBoxImpl$withCompilerApi$api$.compiler$lzycompute(ToolBoxFactory.scala:331)
[error]     	at scala.tools.reflect.ToolBoxFactory$ToolBoxImpl$withCompilerApi$api$.compiler(ToolBoxFactory.scala:330)
[error]     	at scala.tools.reflect.ToolBoxFactory$ToolBoxImpl$withCompilerApi$.liftedTree2$1(ToolBoxFactory.scala:357)
[error]     	at scala.tools.reflect.ToolBoxFactory$ToolBoxImpl$withCompilerApi$.apply(ToolBoxFactory.scala:355)
[error]     	at scala.tools.reflect.ToolBoxFactory$ToolBoxImpl.compile(ToolBoxFactory.scala:422)
[error]     	at scala.tools.reflect.ToolBoxFactory$ToolBoxImpl.eval(ToolBoxFactory.scala:444)
[error]     	at scala.reflect.macros.contexts.Evals$class.eval(Evals.scala:20)
[error]     	at scala.reflect.macros.contexts.Context.eval(Context.scala:6)
```

To reproduce, 
```
cd target 
git clone https://github.com/olafurpg/fastparse.git
git checkout 8fad5baecf8558ded1f81d9d783be1a096cf2213 # branch 'native'
sbt fastparseNative/compile
```
I am unable to reproduce the error with `fastparseJS/compile` or `fastparseJVM/compile`.

I've managed to compile fastparse before with native in https://github.com/scala-native/scala-native/issues/557 so this may be a regression from then.

**Comment by densh** [comment]
Thanks for the report!

**Comment by densh** [comment]
Here is a PR that fixes this: #902

We'll release this fix as part of 0.3.2.
